internal/attestation/azure/snp/validator.go

```diff
@@ -52,235 +55,157 @@ func NewValidator(cfg *config.AzureSEVSNP, log attestation.Logger) *Validator {
-	ctx context.Context, cert *x509.Certificate, report snpAttestationReport, maaToken string, extraData []byte,
-) error {
-	if report.Policy.Debug() {
-		return errDebugEnabled
-	}
-
-	if !report.CommittedTCB.isVersion(v.config.BootloaderVersion.Value, v.config.TEEVersion.Value, v.config.SNPVersion.Value, v.config.MicrocodeVersion.Value) {
-		return &versionError{"COMMITTED_TCB", report.CommittedTCB}
-	}
-	if report.LaunchTCB != report.CommittedTCB {
-		return &versionError{"LAUNCH_TCB", report.LaunchTCB}
-	}
-	if !report.CommittedTCB.supersededBy(report.CurrentTCB) {
-		return &versionError{"CURRENT_TCB", report.CurrentTCB}
-	}
-
-	if err := validateVCEKExtensions(cert, report); err != nil {
-		return fmt.Errorf("mismatching vcek extensions: %w", err)
-	}
-
-	sigR := report.Signature.R[:]
-	sigS := report.Signature.S[:]
-
-	// Table 107 in https://www.amd.com/system/files/TechDocs/56860.pdf mentions little endian signature components.
-	// They come out of the certificate as big endian.
-	reverseEndian(sigR)
-	reverseEndian(sigS)
-
-	rParam := new(big.Int).SetBytes(sigR)
-	sParam := new(big.Int).SetBytes(sigS)
-	sequence := ecdsaSig{rParam, sParam}
-	sigEncoded, err := asn1.Marshal(sequence)
-	if err != nil {
-		return fmt.Errorf("marshalling ecdsa signature: %w", err)
-	}
-
-	buf := new(bytes.Buffer)
-	if err = binary.Write(buf, binary.LittleEndian, report); err != nil {
-		return fmt.Errorf("writing report to buf: %w", err)
-	}
-	// signature is only calculated from 0x0 to 0x2a0
-	if err := cert.CheckSignature(x509.ECDSAWithSHA384, buf.Bytes()[:0x2a0], sigEncoded); err != nil {
-		return &signatureError{err}
-	}
-
+// checkIDKeyDigest validates the IDKeyDigest in the given attestation report against the accepted IDKeyDigests in the
+// validator's config. If an IDKeyDigest is present in the report that is not in the accepted IDKeyDigests, the validation proceeds
+// according to the enforcement policy. If the enforcement policy is set to MAAFallback, the maaToken is validated against the MAA.
+// If the enforcement policy is set to WarnOnly, a warning is logged. If the enforcement policy is set to neither WarnOnly or MAAFallback, an
+// error is returned.
+func (v *Validator) checkIDKeyDigest(ctx context.Context, report *spb.Attestation, maaToken string, extraData []byte) error {
 	hasExpectedIDKeyDigest := false
 	for _, digest := range v.config.FirmwareSignerConfig.AcceptedKeyDigests {
-		if bytes.Equal(digest, report.IDKeyDigest[:]) {
+		if bytes.Equal(digest, report.Report.IdKeyDigest) {
 			hasExpectedIDKeyDigest = true
 			break
 		}
 	}
 
 	if !hasExpectedIDKeyDigest {
+		// IDKeyDigests that were not expected are present, check the enforcement policy and verify against
+		// the MAA if necessary.
 		switch v.config.FirmwareSignerConfig.EnforcementPolicy {
 		case idkeydigest.MAAFallback:
 			v.log.Infof(
 				"configured idkeydigests %x don't contain reported idkeydigest %x, falling back to MAA validation",
```

Comment:
```suggestion
				"Configured idkeydigests %x don't contain reported idkeydigest %x, falling back to MAA validation",
```
Log messages should start in upper case